From: Kerin Millar <kerframil@gmail.com>

Fix build error where PAX_HOOK_ACL_FLAGS is enabled along with 32-bit 
ELF support on x86_64/ia64 platforms. Closes gentoo bug 208331.

--- a/fs/binfmt_elf.c  2008-02-09 00:01:18.000000000 +0100
+++ b/fs/binfmt_elf.c  2008-03-08 01:49:25.000000000 +0100
@@ -47,11 +47,6 @@
 #include <asm/desc.h>
 #endif

-#ifdef CONFIG_PAX_HOOK_ACL_FLAGS
-void (*pax_set_initial_flags_func)(struct linux_binprm *bprm);
-EXPORT_SYMBOL(pax_set_initial_flags_func);
-#endif
-
 static int load_elf_binary(struct linux_binprm *bprm, struct pt_regs *regs);
 static int load_elf_library(struct file *);
 static unsigned long elf_map (struct file *, unsigned long, struct elf_phdr *, int, int);
--- a/fs/exec.c        2008-02-09 00:01:18.000000000 +0100
+++ b/fs/exec.c        2008-03-08 01:49:02.000000000 +0100
@@ -61,6 +61,11 @@
 #include <linux/kmod.h>
 #endif

+#ifdef CONFIG_PAX_HOOK_ACL_FLAGS
+void (*pax_set_initial_flags_func)(struct linux_binprm *bprm);
+EXPORT_SYMBOL(pax_set_initial_flags_func);
+#endif
+
 int core_uses_pid;
 char core_pattern[CORENAME_MAX_SIZE] = "core";
 int suid_dumpable = 0;